plan/logical_plan_builder.go

```diff
@@ -872,8 +874,21 @@ func (b *planBuilder) buildTableDual() LogicalPlan {
 }
 
 func (b *planBuilder) getTableStats(table *model.TableInfo) *statistics.Table {
-	// TODO: Currently we always return a pseudo table for good performance. We will use a cache in future.
-	return statistics.PseudoTable(table)
+	txn := b.ctx.Txn()
+	if txn == nil {
+		return statistics.PseudoTable(table)
+	}
+	m := meta.NewMeta(txn)
+	tblStats, err := m.GetTableStats(table.ID)
```

Comment:
This operation is expensive, may be called multiple times for a single statement.
We should implement a cache, otherwise, the performance of small query would be much slower.